### ISSUE TITLE
Extend Forums globalUnionSearch()

### ISSUE THREAD

**Issue by Bjoern3003** [PHPfox-Official/phpfox-v4-feature-requests#848]
To search also in posts Text and not only in Forum Title.

`$this->database()->select('item.thread_id AS item_id, item.title AS item_title, item.time_stamp AS item_time_stamp, item.user_id AS item_user_id, \'forum\' AS item_type_id, \'\' AS item_photo, 0 AS item_photo_server')
            ->from(Phpfox::getT('forum_thread'), 'item')
            ->leftJoin(Phpfox::getT('forum_post'), 'post', 'post.thread_id = item.thread_id')
            ->leftJoin(Phpfox::getT('forum_post_text'), 'post_text', 'post_text.post_id = post.post_id')
            ->where('item.view_id = 0 AND (' . $this->database()->searchKeywords('item.title',
                    $sSearch) . ' OR ' . $this->database()->searchKeywords('post_text.text',
                    htmlentities($sSearch)) . ') AND item.group_id=0')
            ->union();`

**Comment by harrison05** [comment]
You can add it on your own app Callback to customize global search on Forums
```php
    public function globalUnionSearch($sSearch)
    {
        .....
    }
    public function getSearchInfo($aRow)
    {
       ....
    }
    public function getSearchTitleInfo()
    {
       ...
    }
```
About request update on Forums. We need to review it.

Regards.

**Comment by Bjoern3003** [comment]
I know about the callback functions.

But I can not customize the functions of forum app. I did not want to change core app files.

**Comment by harrison05** [comment]
I mean, you can add those callback in your app to search in Forum Post table, using union() query.

**Comment by Bjoern3003** [comment]
Did you have an example for me to understand? 

If I click in the search Page on the left side on forum, I did not get the result?! Because he is using the callbacks from forum app only . Is that right?

**Comment by harrison05** [comment]
> If I click in the search Page on the left side on forum, I did not get the result?! Because he is using the callbacks from forum app only . Is that right?

Yes, it only search forum app. But you can add menu to left side in your App, example callback:
```php
//Write this code in your app callback
public function getSearchTitleInfo()
    {
        return array(
            'name' => _p('forum_posts')
        );
    }
```
when user click on that menu, it will call to function **globalUnionSearch** in your app.

**Comment by Bjoern3003** [comment]
This is not, what I want to Feature Request with this post. 

I want, that the global search also seach keywords in Forum Threads, not only in the title.

**Comment by harrison05** [comment]
> This is not, what I want to Feature Request with this post.
> 
> I want, that the global search also seach keywords in Forum Threads, not only in the title.

Oh, I got this. We will review your request.

Regards.

**Comment by Bjoern3003** [comment]
Is there any update status?
